code/entities/machine/soda-fountain/SodaFountain.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Editor;
+using Sandbox;
+using Sandbox.util;
+
+namespace Cinema;
+
+[Library("ent_soda_fountain"), HammerEntity]
+[Title("Soda Fountain"), Category("Gameplay"), Icon("coffee_maker")]
+[EditorModel("models/sodafountain/sodafountain_01.vmdl")]
+public partial class SodaFountain : AnimatedEntity, ICinemaUse
+{
+    public enum SodaType : int
+    {
+        Conk = 0,
+        MionPisz = 1,
+        Spooge = 2
+    }
+    public enum CupColor : int
+    {
+        White = 0,
+        Blue = 1,
+        Green = 2,
+        Red = 3,
+        Black = 4
+    }
+
+    [Net] public IList<BaseInteractable> Interactables { get; set; }
+
+    /// <summary>
+    /// Set up the model when spawned by the server
+    /// Setup model
+    /// Setup interactions
+    /// </summary>
+    public override void Spawn()
+    {
+        base.Spawn();
+
+        SetModel("models/sodafountain/sodafountain_01.vmdl");
+
+        SetupPhysicsFromModel(PhysicsMotionType.Keyframed);
+
+        AddInteractables();
+        
+        Tags.Add("interactable");
+    }
+
+    /// <summary>
+    /// Plays soda fountain sound on client spawn
+    /// </summary>
+    public override void ClientSpawn()
+    {
+        base.ClientSpawn();
+
+        Sound.FromEntity("machine_hum_02", this).SetVolume(8.0f);
```

Comment:
Does this sound loop? How does this sound stop once the entity is destroyed?

Reply:
The sound loops. It seems to get destroyed when the entity is deleted as well.